docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,179 @@
+
+    Upgrade to the latest stable version:
+
+    COMPONENT            CURRENT   AVAILABLE
+    API Server           v1.8.3    v1.9.2
+    Controller Manager   v1.8.3    v1.9.2
+    Scheduler            v1.8.3    v1.9.2
+    Kube Proxy           v1.8.3    v1.9.2
+    Kube DNS             1.14.5    1.14.7
+    Etcd                 3.2.7     3.1.11
+
+Note that upgrading etcd is not supported here because in the supported configuration it is running externally, hence if necessary you'll have to upgrade it according to etcd's upgrade instruction which is beyond scope here.
+
+## Upgrading your control plane
+
+The following procedure is applied on all master nodes one by one, i.e. when it has finished on master _#1_ it is applied on master _#2_ etc.
+
+Before the actual upgrade `configmap/kubeadm-config` needs to be adapted - replace a hard reference to some master host by the current master host’s FQDN:
+
+```shell
+$ kubectl get configmap -n kube-system kubeadm-config -o yaml >/tmp/kubeadm-config-cm.yaml
+$ sed -i 's/^\([ \t]*nodeName:\).*/\1 <CURRENT-MASTER-FQDN>/' /tmp/kubeadm-config-cm.yaml
+$ kubectl apply -f /tmp/kubeadm-config-cm.yaml --force
+```
+
+Now the actual upgrade process can start. Use the target version determined in the preparation step and run the following command (press “y” when prompted):
+
+```shell
+$ /tmp/kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+If the operation was successful you’ll get a message like this:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.2". Enjoy!
+
+    To upgrade the cluster with CoreDNS as the default internal DNS, invoke `kubeadm upgrade apply` with the `--feature-gates=CoreDNS=true` flag.
+
+`kubeadm upgrade apply` does the following:
+
+- Checks that your cluster is in an upgradeable state:
+  - The API server is reachable,
+  - All nodes are in the `Ready` state
+  - The control plane is healthy
+- Enforces the version skew policies.
+- Makes sure the control plane images are available or available to pull to the machine.
+- Upgrades the control plane components or rollbacks if any of them fails to come up.
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
+- Creates new certificate and key files of apiserver and backs up old files if they're about to expire in 180 days.
+
+Next, manually upgrade your Software Defined Network (SDN).
+
+Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow. Check the [addons](/docs/concepts/cluster-administration/addons/) page to find your CNI provider and see if there are additional upgrade steps necessary.
+
+It has been observed that in some cases after having upgraded the primary master the upgrade on some of the secondary masters failed initially (timed out waiting for the restarted static pods to come up) but worked just fine when retried after a minute or two.
```

Comment:
"It has been observed" is a very passive phrasing - is there a way to make this more active voice? Is the end result "don't worry - just try again and it's OK"? I wasn't quite sure myself when reading the walk-through